docs/src/DeveloperGuide/WritingAdaptor.md

```diff
@@ -0,0 +1,264 @@
+# Tutorial: Writing a new adaptor
+
+In this tutorial we will develop a new QAT profile adaptor. We will make the
+adaptor a separate library which is dynamically loaded through the command line
+interface. All examples in this tutorial can be found in `AdaptorExamples`.
+
+Our first "adaptor" will be a "hello world" boilerplate which serves the purpose
+of giving the reader an understanding of how to define configurations for our
+adaptor. We will demonstrate how to use this adaptor from the command line.
+
+For our second adaptor, we will use a standard LLVM pass to demonstrate how to
+load these. We will show how the registered configuration can be used to enable
+or disable the pass. To show that the effect of the pass, we use the inliner
+pipeline. We will see how enabling the pass results in inlining all the function
+calls. As QAT ships with a built-in inliner pass, it is important to remember to
+disable this to see the effect of our custom pass.
```

Comment:
This last sentence feels a bit odd here; it fits/serves better to state this in the corresponding section instead.

Reply:
This sentence is a left-over from the old docs. It is not needed since it is automatically disabled when we specify a new pipeline.